### PR TITLE
Support mixtral moe AWQ quantization.

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

**Comment by lvhan028** [comment]
@zhulinJulia24 please add mistralai/Mixtral-8x7B-Instruct-v0.1 awq quantization into test cases

File: lmdeploy/lite/quantization/awq.py [hunk 1]
```diff
@@ -244,6 +254,9 @@ def quant_weights(model,
         if skip_if_contains and skip_if_contains in child_name:
             q_linear = fc
             pack_or_skip = 'skipped'
+        elif 'block_sparse_moe.gate' in name:  # moe
```
Comment (lvhan028) [inline lmdeploy/lite/quantization/awq.py:257]: This is an additional skip patch, considering we already possess the `skip_if _contains` functionality. There is a growing concern that as more Mixtures of Experts (MoEs) are integrated, the skipping branches may become increasingly difficult to maintain.

**Comment by AllentDan** [comment]
模型访问不了。可以断点进 observer.py 108 行，应该是模型推理过程某个层前后有个 tensor shape 不对

**Review by lvhan028 — APPROVED** [review]

**Comment by anaivebird** [comment]
Error when awq quantize mistralai/Mixtral-8x7B-Instruct-v0.1 model @AllentDan 

```
Traceback (most recent call last):
  File "/opt/py38/bin/lmdeploy", line 11, in <module>
    load_entry_point('lmdeploy', 'console_scripts', 'lmdeploy')()
  File "/workspace/xingwu/lmdeploy/lmdeploy/cli/entrypoint.py", line 42, in run
    args.run(args)
  File "/workspace/xingwu/lmdeploy/lmdeploy/cli/lite.py", line 131, in auto_awq
    auto_awq(**kwargs)
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/apis/auto_awq.py", line 91, in auto_awq
    vl_model, model, tokenizer, work_dir = calibrate(model,
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/apis/calibrate.py", line 319, in calibrate
    calib_ctx.calibrate(all_data)
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/quantization/calibration.py", line 238, in calibrate
    _ = model(data.to(self.device))
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/transformers/models/mixtral/modeling_mixtral.py", line 1002, in forward
    layer_outputs = decoder_layer(
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/quantization/calibration.py", line 457, in _forward
    auto_scale_block(mod, batch_kwargs[i], self.w_bits,
  File "/opt/py38/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/quantization/calibration.py", line 349, in auto_scale_block
    _auto_get_scale(
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/quantization/calibration.py", line 331, in _auto_get_scale
    best_ratio = _search_module_scale(module2inspect, layers, inp.value,
  File "/workspace/xingwu/lmdeploy/lmdeploy/lite/quantization/calibration.py", line 278, in _search_module_scale
    org_out = block(x, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/opt/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
TypeError: forward() got an unexpected keyword argument 'output_router_logits'
```


```
export CUDA_VISIBLE_DEVICES=2,3
export HF_MODEL=/workspace/models/mixtral-moe
export WORK_DIR=/workspace/models/mixtral-moe-4bit

lmdeploy lite auto_awq \
   $HF_MODEL \
  --calib-dataset 'ptb' \
  --calib-samples 128 \
  --calib-seqlen 2048 \
  --w-bits 4 \
  --w-group-size 128 \
  --batch-size 1 \
  --search-scale False \
  --work-dir $WORK_DIR
```

**Comment by AllentDan** [comment]
@anaivebird Remove `--search-scale False` please.

**Comment by anaivebird** [comment]
> vl_model, model, tokenizer, work_dir = calibrate(model,

Thanks, it works.  But remove `--search-scale False` should equivalent to add `--search-scale False`, in both case, search-scale is False. Why the program result in different path?

**Comment by AllentDan** [comment]
It was a bug

**Comment by lvhan028** [comment]
@AllentDan w4a16.md, quantization_utils.py should be updated due to the change of `--search-scale`

**Pull request merged**